src/transformers/lib_integrations/peft/peft_mixin.py

```diff
@@ -125,28 +131,32 @@ def load_adapter(
         elif adapter_name in self.peft_config:
             raise ValueError(f"Adapter with name {adapter_name} already exists. Please use a different name.")
 
-        adapter_config_file = find_adapter_config_file(
-            peft_model_id,
-            revision=revision,
-            token=token,
-        )
-
-        if adapter_config_file is None:
-            raise ValueError(
-                f"adapter model file not found in {peft_model_id}. Make sure you are passing the correct path to the "
-                "adapter model."
+        if peft_config is None:
```

Comment:
```suggestion
        if peft_config is None and isinstance(peft_model_id, str):
```
and add a check below that if peft_model_id is not a state dict, raise an error